### PR TITLE
Updating automatic changelog generation script

### PR DESCRIPTION
**Pull request by mohammedzee1000**
Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

**What type of PR is this?**
/kind cleanup

**What does this PR do / why we need it**:
This PR adds updated changelog script for automatic generation of changelog during release

**How this works**:
```bash
> ./scripts/changelog-script.sh v2.2.3 v2.2.4
Ignoring bigdecimal-3.0.0 because its extensions are not built. Try: gem pristine bigdecimal --version 3.0.0
Ignoring date-3.1.1 because its extensions are not built. Try: gem pristine date --version 3.1.1
Ignoring etc-1.2.0 because its extensions are not built. Try: gem pristine etc --version 1.2.0
Ignoring ffi-1.15.0 because its extensions are not built. Try: gem pristine ffi --version 1.15.0
Ignoring io-console-0.5.8 because its extensions are not built. Try: gem pristine io-console --version 0.5.8
Ignoring json-2.5.1 because its extensions are not built. Try: gem pristine json --version 2.5.1
Ignoring json-2.3.0 because its extensions are not built. Try: gem pristine json --version 2.3.0
Ignoring openssl-2.2.0 because its extensions are not built. Try: gem pristine openssl --version 2.2.0
Ignoring psych-3.3.1 because its extensions are not built. Try: gem pristine psych --version 3.3.1
Ignoring racc-1.5.2 because its extensions are not built. Try: gem pristine racc --version 1.5.2
Ignoring stringio-3.0.0 because its extensions are not built. Try: gem pristine stringio --version 3.0.0
Ignoring strscan-3.0.0 because its extensions are not built. Try: gem pristine strscan --version 3.0.0
Using these options:
:date_format=>"%Y-%m-%d"
:output=>"/tmp/changelog"
:base=>"/tmp/base"
:issues=>false
:add_issues_wo_labels=>true
:add_pr_wo_labels=>true
:pulls=>true
:filter_issues_by_milestone=>true
:author=>true
:unreleased=>true
:unreleased_label=>"Unreleased"
:compare_link=>true
:exclude_labels=>["lifecycle/rotten", "duplicate", "question", "invalid", "wontfix"]
:summary_labels=>["Release summary", "release-summary", "Summary", "summary"]
:breaking_labels=>["backwards-incompatible", "Backwards incompatible", "breaking"]
:enhancement_labels=>["kind/feature"]
:bug_labels=>["kind/bug"]
:deprecated_labels=>["deprecated", "Deprecated", "Type: Deprecated"]
:removed_labels=>["removed", "Removed", "Type: Removed"]
:security_labels=>["security", "Security", "Type: Security"]
:configure_sections=>{}
:add_sections=>"{\"documentation\":{\"prefix\":\"**Documentation:**\",\"labels\":[\"kind/documentation\"]}, \"tests\": {\"prefix\": \"**Testing/CI:**\", \"labels\": [\"kind/tests\"]}, \"cleanup\": {\"prefix\": \"**Cleanup/Refactor:\", \"labels\": [\"kind/cleanup\"]}}"
:issue_line_labels=>[]
:max_issues=>500
:simple_list=>false
:ssl_ca_file=>nil
:verbose=>true
:header=>"# Release of v2.2.4"
:merge_prefix=>"**Merged pull requests:**"
:issue_prefix=>"**Closed issues:**"
:summary_prefix=>""
:breaking_prefix=>"**Breaking changes:**"
:enhancement_prefix=>"**Features/Enhancements:**"
:bug_prefix=>"**Bugs:**"
:deprecated_prefix=>"**Deprecated:**"
:removed_prefix=>"**Removed:**"
:security_prefix=>"**Security fixes:**"
:http_cache=>true
:require=>[]
:user=>"openshift"
:project=>"odo"
:token=>"hidden value"
:since_tag=>"v2.2.3"
:future_release=>"v2.2.4"

Found 59 tags                                                          
Fetching tags dates: 59/59
Sorting tags...
Received issues: 500                                                   
Pull Request count: 2429                                               
Filtered pull requests: 279
Fetching events for issues and PR: 279                                 
Fetching closed dates for issues: 279/279
Fetching SHAs for tags: 59                                             
Warning: PR 4749 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 4725 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 4718 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 4696 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 4676 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 4672 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 4611 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Warning: PR 4417 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
Associating PRs with tags: 279/279
Generating entry...
Done!
Generated log placed in /home/mzee1000/workbench/work/golang/redhat-gomod/odo//tmp/changelog

The changelog is located at: /tmp/changelog
> cat /tmp/changelog
# Release of v2.2.4

## [v2.2.4](https://github.com/openshift/odo/tree/v2.2.4) (2021-08-18)

[Full Changelog](https://github.com/openshift/odo/compare/v2.2.4...v2.2.4)

**Features/Enhancements:**

- Adds a column in odo list output to indicate components created by odo [\#4962](https://github.com/openshift/odo/pull/4962) ([mik-dass](https://github.com/mik-dass))

**Documentation:**

- Remove empty docs; add when ready [\#4986](https://github.com/openshift/odo/pull/4986) ([dharmit](https://github.com/dharmit))
- Documents steps for component creation [\#4982](https://github.com/openshift/odo/pull/4982) ([dharmit](https://github.com/dharmit))

**Cleanup/Refactor:

- Add deprecation warnings for s2i components [\#4967](https://github.com/openshift/odo/pull/4967) ([valaparthvi](https://github.com/valaparthvi))
- odo link: fix command description [\#4930](https://github.com/openshift/odo/pull/4930) ([kadel](https://github.com/kadel))
- Cleanup cmd\_link\_unlink\_test [\#4891](https://github.com/openshift/odo/pull/4891) ([valaparthvi](https://github.com/valaparthvi))

**Merged pull requests:**

- Updating pr template [\#5003](https://github.com/openshift/odo/pull/5003) ([mohammedzee1000](https://github.com/mohammedzee1000))
- Adds missing \# to a Changelog entry [\#4983](https://github.com/openshift/odo/pull/4983) ([dharmit](https://github.com/dharmit))
- Updated npm modules for Dependabot warnings [\#4977](https://github.com/openshift/odo/pull/4977) ([dharmit](https://github.com/dharmit))
- Fix reports directory for operatorhub nightly tests [\#4972](https://github.com/openshift/odo/pull/4972) ([prietyc123](https://github.com/prietyc123))

## [v2.2.4](https://github.com/openshift/odo/tree/v2.2.4) (2021-08-04)

[Full Changelog](https://github.com/openshift/odo/compare/v2.2.3...v2.2.4)

**Features/Enhancements:**

- Nightly CI and Operator Setup script [\#4958](https://github.com/openshift/odo/pull/4958) ([prietyc123](https://github.com/prietyc123))
- Do not cancel the whole execution if one of the test suite fails [\#4943](https://github.com/openshift/odo/pull/4943) ([prietyc123](https://github.com/prietyc123))
- preStart events support [\#4909](https://github.com/openshift/odo/pull/4909) ([feloy](https://github.com/feloy))
- Adds support for linking using service binding without the service binding operator [\#4905](https://github.com/openshift/odo/pull/4905) ([mik-dass](https://github.com/mik-dass))
- Add Sonar Cloud config file: sonar-project.properties [\#4900](https://github.com/openshift/odo/pull/4900) ([valaparthvi](https://github.com/valaparthvi))
- Add linked service information in odo describe [\#4866](https://github.com/openshift/odo/pull/4866) ([feloy](https://github.com/feloy))

**Bugs:**

- Test storage + link [\#4954](https://github.com/openshift/odo/pull/4954) ([feloy](https://github.com/feloy))
- Modifies the error message to list the available starters [\#4935](https://github.com/openshift/odo/pull/4935) ([mik-dass](https://github.com/mik-dass))
- Do not display "Waiting for component to start" twice [\#4919](https://github.com/openshift/odo/pull/4919) ([feloy](https://github.com/feloy))
- Add language and projectType metadata in devfile for s2i components [\#4911](https://github.com/openshift/odo/pull/4911) ([valaparthvi](https://github.com/valaparthvi))
- Update component create and delete help examples [\#4904](https://github.com/openshift/odo/pull/4904) ([valaparthvi](https://github.com/valaparthvi))
- Fixing now implementation for config set and unset in devfile [\#4882](https://github.com/openshift/odo/pull/4882) ([mohammedzee1000](https://github.com/mohammedzee1000))
- Renames the service name to include the app name [\#4858](https://github.com/openshift/odo/pull/4858) ([mik-dass](https://github.com/mik-dass))

**Documentation:**

- Add odo installation docs [\#4923](https://github.com/openshift/odo/pull/4923) ([valaparthvi](https://github.com/valaparthvi))
- Add documentation for global configuration [\#4921](https://github.com/openshift/odo/pull/4921) ([valaparthvi](https://github.com/valaparthvi))
- Add Getting Started \> Basics documentation [\#4914](https://github.com/openshift/odo/pull/4914) ([valaparthvi](https://github.com/valaparthvi))

**Cleanup/Refactor:

- Remove unused Docker code [\#4947](https://github.com/openshift/odo/pull/4947) ([feloy](https://github.com/feloy))
- Updating dependencies to address dependabot alerts [\#4912](https://github.com/openshift/odo/pull/4912) ([dharmit](https://github.com/dharmit))

**Merged pull requests:**

- Bumping odo version before release [\#4966](https://github.com/openshift/odo/pull/4966) ([mohammedzee1000](https://github.com/mohammedzee1000))
- Removes Girish from OWNERS files [\#4926](https://github.com/openshift/odo/pull/4926) ([dharmit](https://github.com/dharmit))
- Refactor devfile catalog and config integration tests [\#4915](https://github.com/openshift/odo/pull/4915) ([feloy](https://github.com/feloy))
- Migrating URL unit tests to use networking v1 ingress [\#4902](https://github.com/openshift/odo/pull/4902) ([mohammedzee1000](https://github.com/mohammedzee1000))
- Enabling override of required golang in rpm [\#4888](https://github.com/openshift/odo/pull/4888) ([mohammedzee1000](https://github.com/mohammedzee1000))
- delete dangling dir created by unit-tests [\#4887](https://github.com/openshift/odo/pull/4887) ([anandrkskd](https://github.com/anandrkskd))
- Bumping version of odo for release [\#4880](https://github.com/openshift/odo/pull/4880) ([mohammedzee1000](https://github.com/mohammedzee1000))
- Tests/fix for test on ibmc [\#4873](https://github.com/openshift/odo/pull/4873) ([anandrkskd](https://github.com/anandrkskd))
- Limits memory requests to deploy more components [\#4869](https://github.com/openshift/odo/pull/4869) ([feloy](https://github.com/feloy))

# Installation of v2.2.4
To install odo, follow our installation guide at [docs.openshift.com](https://docs.openshift.com/container-platform/latest/cli_reference/developer_cli_odo/installing-odo.html)
After each release, binaries are synced to [mirror.openshift.com](https://mirror.openshift.com/pub/openshift-v4/clients/odo/v2.2.4/)


\* *This Changelog was automatically generated by [github_changelog_generator](https://github.com/github-changelog-generator/github-changelog-generator)*
```

### CODE REVIEW

File: scripts/changelog-script.sh [hunk 1]
```diff
@@ -12,26 +22,29 @@ fi
...
+--max-issues 500 \
+--user openshift \
+--project odo \
+--no-issues \
+-t $GITHUB_TOKEN \
+--since-tag $1 \
+--future-release $2 \
+--base /tmp/base \
+--output /tmp/changelog \
+--exclude-labels "lifecycle/rotten,duplicate,question,invalid,wontfix" \
```
Comment (dharmit) [inline scripts/changelog-script.sh:40]: I don't think we have these labels in odo repo: `duplicate`, `question`, `invalid`, `wontfix`. Do we?

**Review by dharmit — COMMENTED** [review]
Just one question. Looks good otherwise.

File: scripts/changelog-script.sh (hunk 1 above)
Comment (mohammedzee1000) [reply scripts/changelog-script.sh:40]: There are the default labels, so i chose to keep them

**Review by dharmit — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**